tap_partnerize/client.py

```diff
@@ -98,33 +158,22 @@ def get_url_params(
         Returns:
             A dictionary of URL query parameters.
         """
-        params: dict = {}
-        if next_page_token:
-            params["page"] = next_page_token
+        params: dict = {
+            "publisher_id": self.config.get("publisher_id", ""),
+            "ref_conversion_metric_id": "2"
+        }
+
+        next_page_date = datetime.datetime.strftime(next_page_token, "%Y-%m-%d")
+        if next_page_date:
+            params["start_date"] = next_page_date
+            end_datetime = datetime.datetime.strptime(next_page_date, "%Y-%m-%d") + datetime.timedelta(days=1)
+            params["end_date"] = datetime.datetime.strftime(end_datetime, "%Y-%m-%d")
+
         if self.replication_key:
             params["sort"] = "asc"
             params["order_by"] = self.replication_key
         return params
 
-    def prepare_request_payload(
-        self,
-        context: dict | None,
-        next_page_token: Any | None,
-    ) -> dict | None:
-        """Prepare the data payload for the REST API request.
-
-        By default, no payload will be sent (return None).
-
-        Args:
-            context: The stream context.
-            next_page_token: The next page index or value.
-
-        Returns:
-            A dictionary with the JSON body for a POST requests.
-        """
-        # TODO: Delete this method if no payload is required. (Most REST APIs.)
-        return None
-
     def parse_response(self, response: requests.Response) -> Iterable[dict]:
```

Comment:
1. are we doing all the post processing in `parse_response()` instead of in `post_process()`? what was the rationale for that?
2. is it worth refactoring some of it out into functions? for example, it would be more readable if the field name cleanup was instead a call to a `make_valid_bigquery_field_name()` function or something

Reply:
There wasn't any real rationale for choosing one over the other. I think I just didn't fully understand at the start how their responsibilities were separate, and once it worked with `parse_response`, I didn't revisit it. I could probably handle the top-level response parsing in `parse_response` and then tweak the field-level results in `post_process`. I'll take a look into that.

Reply:
here's my mental model for this:
* `parse_response()` is responsible just for taking in the API response and returning rows of data. for example, if the response has a `data` attribute with a 100 items it's responsible for returning each of those 100 as separate rows, but without changing the data
* `post_process()` then consumes each of those rows and performs any necessary cleanup or transformation. even this our convention is to do the bare minimum needed to get this into BQ in a form we can work with, and then do all further processing in SQL w/ dbt